x-pack/plugins/security_solution/public/management/pages/response_actions/view/components/action_list_date_range_picker.tsx

```diff
@@ -0,0 +1,105 @@
+import { useUiSetting$ } from '@kbn/kibana-react-plugin/public';
+import { DurationRange, OnRefreshChangeProps } from '@elastic/eui/src/components/date_picker/types';
+import { useKibana } from '../../../../../common/lib/kibana/kibana_react';
+import { DEFAULT_TIMEPICKER_QUICK_RANGES } from '../../../../../../common/constants';
+
+export interface DateRangePickerValues {
+  autoRefreshOptions: {
+    enabled: boolean;
+    duration: number;
+  };
+  startDate?: string;
+  endDate?: string;
+  recentlyUsedDateRanges: EuiSuperDatePickerRecentRange[];
+}
+interface Range {
+  from: string;
+  to: string;
+  display: string;
+}
+
+const DatePickerWrapper = euiStyled.div`
+  width: ${(props) => props.theme.eui.fractions.single.percentage};
+`;
+const StickyFlexItem = euiStyled(EuiFlexItem).attrs({ grow: false })`
+  background: ${(props) => `${props.theme.eui.euiHeaderBackgroundColor}`};
+  position: sticky;
+  top: 0;
+  z-index: 1;
+  padding: ${(props) => `${props.theme.eui.euiSizeL}`};
+  padding-left: ${(props) => `${props.theme.eui.euiSizeM}`};
+`;
+
+export const ActionListDateRangePicker = memo(
+  ({
+    dateRangePickerState,
+    isDataLoading,
+    onRefresh,
+    onRefreshChange,
+    onTimeChange,
+  }: {
+    dateRangePickerState: DateRangePickerValues;
+    isDataLoading: boolean;
+    onRefresh: () => void;
+    onRefreshChange: (evt: OnRefreshChangeProps) => void;
+    onTimeChange: ({ start, end }: DurationRange) => void;
+  }) => {
+    const { uiSettings } = useKibana().services;
+    const [quickRanges] = useUiSetting$<Range[]>(DEFAULT_TIMEPICKER_QUICK_RANGES);
+    const [dateFormat] = useState(() => uiSettings?.get('dateFormat'));
+    const commonlyUsedRanges = !quickRanges.length
+      ? []
+      : quickRanges.map(({ from, to, display }) => ({
+          start: from,
+          end: to,
+          label: display,
+        }));
+
+    return (
+      <StickyFlexItem>
+        <EuiFlexGroup justifyContent="flexStart" responsive>
+          <DatePickerWrapper data-test-subj="activityLogSuperDatePicker">
+            <EuiFlexItem>
+              <EuiSuperDatePicker
+                updateButtonProps={{ iconOnly: true, fill: false }}
+                isLoading={isDataLoading}
+                dateFormat={dateFormat}
+                commonlyUsedRanges={commonlyUsedRanges}
+                end={
+                  dateRangePickerState.endDate
+                    ? dateMath.parse(dateRangePickerState.endDate)?.toISOString()
+                    : undefined
+                }
+                isPaused={!dateRangePickerState.autoRefreshOptions.enabled}
+                onTimeChange={onTimeChange}
+                onRefreshChange={onRefreshChange}
+                refreshInterval={dateRangePickerState.autoRefreshOptions.duration}
+                onRefresh={onRefresh}
+                recentlyUsedRanges={dateRangePickerState.recentlyUsedDateRanges}
+                start={
+                  dateRangePickerState.startDate
```

Comment:
Can you put this in a `useMemo`?

Reply:
dd244ae3f28b5a2945b139e9f2f21344e1a4c7a9